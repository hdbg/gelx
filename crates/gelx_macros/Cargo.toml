[package]
name = "gelx_macros"
version = { workspace = true }
authors = { workspace = true }
categories = ["database"]
edition = { workspace = true }
homepage = { workspace = true }
keywords = ["gel", "database", "typesafe", "checked", "macros"]
license = { workspace = true }
readme = "readme.md"
repository = { workspace = true }
rust-version = { workspace = true }
description = "The proc-macro crate for the `gelx` crate."

[lib]
proc-macro = true

[dependencies]
gelx_core = { workspace = true }
proc-macro2 = { workspace = true }
quote = { workspace = true }
syn = { workspace = true }

[features]
query = ["gelx_core/query"]
serde = ["gelx_core/serde"]
builder = ["gelx_core/builder"]
strum = ["gelx_core/strum"]

[lints]
workspace = true
