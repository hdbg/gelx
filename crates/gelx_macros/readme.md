# `gelx_macros`

<br />

> This crate contains the raw macros for the `gelx` crate.
