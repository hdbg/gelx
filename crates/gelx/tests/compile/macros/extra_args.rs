use gelx_macros::gelx_raw;

fn main() {
	gelx_raw!(example, query: "select bool", "more stuff", that is, "not allowed");
}
