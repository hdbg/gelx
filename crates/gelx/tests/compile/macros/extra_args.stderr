error: unexpected token
 --> tests/compile/macros/extra_args.rs:4:49
  |
4 |     gelx_raw!(example, query: "select bool", "more stuff", that is, "not allowed");
  |                                                    ^
